core/src/main/java/org/infinispan/encoding/DataConversion.java

```diff
@@ -143,30 +142,11 @@ private MediaType getStorageMediaType(Configuration configuration, boolean embed
       return MediaType.APPLICATION_UNKNOWN;
    }
 
-   public boolean isConversionSupported(MediaType mediaType) {
```

Comment:
Can't remove these methods directly IMO, even if we don't expect applications to actually use them they're still part of the public API. Let's deprecate them, but simplify the implementation to simply delegate to `EncoderRegistry`.

Reply:
The problem is to keep them the encoder registry in needed inside the data conversion, which defeats the whole purpose of the the PR :)

Reply:
We can either: 
* deprecate them in the upcoming 10.1.6 and remove here
* just remove it here and write in the upgrade guide

Reply:
IMO if we deprecate those methods and we never call them internally then it's safe to not inject the `EncoderRegistry` in those static instances (i.e. move the `encoderRegistry` assignment after the `if` line in `injectDependencies()`). E.g. https://github.com/infinispan/infinispan/commit/0ffe0955a415cef15e99ed06b9428c6c789b303e

Then everything works as it should, only stuff like `DataConversion.IDENTITY_KEY.convert(o, randomMediaType1, randomMediaType2)`, which shouldn't be a surprise for anyone.

I'm now working on ISPN-11629 and I'm thinking of deprecating those constants too, and using them only internally (if at all). 

OTOH I just noticed that `org.infinispan.encoding` and `org.infinispan.commons.encoding` are not `@public` packages. They are referenced in public APIs like `AdvancedCache`, so they should be in the javadoc. I don't think this gives a free pass to remove methods, but @tristantarrant might disagree :)

Reply:
I'd personally deprecated in 10.1.7 and remove then altogether on 11

Reply:
WDYT @danberindei?

Reply:
@gustavonalle we can't deprecate in 10.1.x any more, so let's just deprecate it in 11.

Reply:
@danberindei updated to simply deprecated it and removed the internal usages